src/schema/document/value.rs

```diff
@@ -259,11 +260,11 @@ impl<'a> ReferenceValueLeaf<'a> {
     }
 
     #[inline]
-    /// If the Value is a pre-tokenized string, returns the associated string. Returns None
-    /// otherwise.
-    pub fn as_pre_tokenized_text(&self) -> Option<Box<PreTokenizedString>> {
+    /// If the Value is a pre-tokenized string, consumes it and returns the string.
+    /// Returns None otherwise.
+    pub fn into_pre_tokenized_text(self) -> Option<Box<PreTokenizedString>> {
```

Comment:
Arguably, all methods on the two reference wrapper types could be `into_*(self)` instead of `as_*(&self)` for consistency but since the others are basically `Copy`, it does not make a difference there.